### PR TITLE
Adding support for yaml configuration file

### PR DESCRIPTION
**Pull request by jay-qrypt**

### CODE REVIEW

File: src/qseed.cpp [hunk 1]
```diff
@@ -24,73 +30,104 @@ struct CommonConfig {
 
 };
 
-CommonConfig getCommonConfig() {
+CommonConfig getCommonConfig(std::string yamlContents) {
```
Comment (jay-qrypt) [inline src/qseed.cpp]: Jay - Make const reference

File: src/qseed.cpp [hunk 2]
```diff
@@ -24,73 +30,104 @@ struct CommonConfig {
...
+        std::string errMsg = "Qseed period must be greater than 0 and less than or equal to " + std::to_string(MAX_QSEED_PERIOD) + ".";
         throw std::runtime_error(errMsg);
     }
 
     return commonConfig;
 
 }
 
-CryptokiConfig getCryptokiConfig() {
+CryptokiConfig getCryptokiConfig(std::string yamlContents) {
```
Comment (jay-qrypt) [inline src/qseed.cpp]: Jay - Make const reference

File: src/qseed.cpp [hunk 3]
```diff
@@ -24,73 +30,104 @@ struct CommonConfig {
...
 
     CryptokiConfig cryptokiConfig = {};
-
-    const char* libraryFile = std::getenv("CRYPTOKI_LIB");
-    if (!libraryFile) {
-        throw std::runtime_error("CRYPTOKI_LIB environment variable is not set");
+    ryml::Tree tree = ryml::parse_in_place((char*)yamlContents.c_str());
+
+    const char* libraryFileKey = "cryptoki_lib";
```
Comment (qryptdev) [inline src/qseed.cpp:73]: Andrew - could optionally create a utility function for getting members from the yaml file given how simple the structure is.

**Review by qryptdev — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -152,12 +165,12 @@ This section covers how to test the application with SoftHSM.
...
 
-4.  Set environment variables. The CRYPTOKI_SLOT_ID should be set to the reassigned slot id from the second step. The CRYPTOKI_USER_PIN should be set to the user PIN from the second step.
-    ```bash
-    export QRYPT_TOKEN=qrypttokenfromportal
-    export CRYPTOKI_LIB=/usr/local/lib/softhsm/libsofthsm2.so
-    export CRYPTOKI_SLOT_ID=824959035
-    export CRYPTOKI_USER_PIN=1234
+4.  Create or update the configuration file at /etc/qseed/qseed_config.yml. The cryptoki_slot_id should be set to the reassigned slot id from the second step. The cryptoki_user_pin should be set to the user PIN from the second step.
+    ```yaml
+    qrypt_token: qrypttokenfromportal
```
Comment (qryptdev) [inline README.md:168]: Andrew - just to check, we don't want size & period here, right?

File: README.md [hunk 5]
```diff
@@ -36,22 +36,24 @@ This section covers how to start the qseed application for Thales Network Luna 7
...
+    | qrypt_token | Token (with Entropy scope) retrieved from the Qrypt portal to get access to Qrypt services. |
+    | size | Amount of seed random in bytes to inject into the HSM at the beginning of each time period. <br>Valid values are inclusively between 1 byte and 65,536 (64 KiB). Defaults to 48. |
+    | period | The time period in seconds between seed random injections. <br>Valid values are inclusively between 1 second and 31,536,000 seconds (about 1 year). Defaults to 10. |
+    | cryptoki_lib | Cryptoki shared library file location. |
+    | cryptoki_slot_id | Cryptoki slot ID as defined in the PKCS11 specification. |
+    | cryptoki_user_pin | Cryptoki crypto user role PIN as defined in the PKCS11 specification. |
+
+    ```yaml
+    qrypt_token: qrypttokenfromportal
+    size: 48      # In bytes
```
Comment (qryptdev) [inline README.md]: Andrew - could consider names that don't require clarifying comments like 'bytes_per_injection' and 'injection_period_seconds', or something similar.

File: README.md (hunk 4 above)
Comment (jay-qrypt) [reply README.md:168]: Correct, they will default to 48 and 10 if not included.

**Review by qryptdev — APPROVED** [review]
Prasanth - LGTM

**Pull request merged**